src/formatDate.js

```diff
@@ -50,7 +50,31 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const initialSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+  const normalizedDate = date.split(initialSeparator);
+  const newFormat = toFormat;
+
+  const dateObject = {
+    'DD': 0,
+    'MM': 0,
+    'YY': 0,
+    'YYYY': 0,
+  };
+
+  for (let item = 0; item < 3; item++) {
+    dateObject[fromFormat[item]] = normalizedDate[item];
+  }
+
+  dateObject['YY'] = dateObject['YY'] === 0
+    ? dateObject['YYYY'].slice(-2) : dateObject['YY'];
```

Comment:
```suggestion
  dateObject['YY'] = dateObject['YY'] === 0
    ? dateObject['YYYY'].slice(-2)
    : dateObject['YY'];
```

Reply:
Formatting is very important
https://mate-academy.github.io/style-guides/javascript.html#comparison--nested-ternaries

Reply:
fixed